File: Programmers/멀쩡한 사각형.cpp
#include <algorithm>
using namespace std;


long long solution(int w, int h) {
    int m = max(w, h);
    int s = min(w, h);
    int tmp = s;
    while ((s % tmp != 0) || (m % tmp != 0)){
        tmp--;
    }
    return (long long)w * h - (w + h - tmp);
}

File: Programmers/멀쩡한 사각형.java
import java.util.*;

class Solution {
    public long solution(int w, int h) {
        long mx = Math.max(w, h);
        long mn = Math.min(w, h);
        long tmp = mx;
        
        while(mx % tmp != 0 || mn % tmp != 0)
            tmp--;

        return mx*mn - (mx + mn - tmp);
    }
}